src/modules/household/components/householdActions/AddRelationshipsStep.tsx

```diff
@@ -0,0 +1,126 @@
+import { Chip, Paper, Stack } from '@mui/material';
+import React, { ReactNode, useId } from 'react';
+import GenericTable from '@/components/elements/table/GenericTable';
+import ClientName from '@/modules/client/components/ClientName';
+import RequiredLabel from '@/modules/form/components/RequiredLabel';
+import HmisEnum from '@/modules/hmis/components/HmisEnum';
+import RelationshipToHohSelect from '@/modules/household/components/elements/RelationshipToHohSelect';
+import { WITH_ENROLLMENT_COLUMNS } from '@/modules/projects/components/tables/ProjectClientEnrollmentsTable';
+import {
+  asClient,
+  CLIENT_COLUMNS,
+} from '@/modules/search/components/ClientSearch';
+import { HmisEnums } from '@/types/gqlEnums';
+import {
+  HouseholdClientFieldsFragment,
+  RelationshipToHoH,
+} from '@/types/gqlTypes';
+
+interface Props {
+  existingClients: HouseholdClientFieldsFragment[];
+  newClients: HouseholdClientFieldsFragment[];
+  relationships: Record<string, RelationshipToHoH | null>;
+  updateRelationship: (
+    enrollmentId: string,
+    relationship: RelationshipToHoH | null
+  ) => void;
+  showNewIndicator?: boolean;
+  children?: ReactNode;
+}
+
+const AddRelationshipsStep = ({
+  existingClients,
+  newClients,
+  relationships,
+  updateRelationship,
+  showNewIndicator,
+  children,
+}: Props) => {
+  const relationshipHeaderId = useId();
+
+  return (
+    <Stack gap={2}>
+      {children}
+      <Paper>
+        <GenericTable<HouseholdClientFieldsFragment>
+          rows={[...existingClients, ...newClients]}
+          columns={[
+            {
+              ...CLIENT_COLUMNS.name,
+              render: (client) => (
+                <Stack direction='row' gap={1}>
+                  <ClientName client={asClient(client)} />
+                  {showNewIndicator && newClients.includes(client) && (
+                    <Chip label='New' size='small' variant='outlined' />
+                  )}
+                </Stack>
+              ),
+              sticky: 'left',
+              tableCellProps: (client) => {
+                return {
+                  // enables using aria-labelledby on inputs in this row
+                  id: `client-${client.id}`,
+                };
+              },
+            },
+            CLIENT_COLUMNS.age,
+            {
+              header: (
+                <RequiredLabel
+                  text='Relationship'
+                  TypographyProps={{
+                    fontWeight: 'bold',
+                  }}
+                  required={true}
+                />
+              ),
+              headerCellProps: {
+                id: relationshipHeaderId,
```

Comment:
Nice. Is it necessary that this ID is on the `th` element itself? I think there is another place where I did something similar with labeling, but the id was inside the th content. The equivalent would be like:
```
              header: (
                <span id={relationshipHeaderId}>
                  <RequiredLabel
                    text='Relationship'
                    TypographyProps={{
                      fontWeight: 'bold',
                    }}
                    required={true}
                  />
                </span>
              ),
```

Would that work just as well for accessible labeling?

Reply:
Yes, I'm pretty sure that would work just as well. I was basing this on the Disability Table approach which I remember having Bobby's stamp of approval :)

I guess the `span` approach would simplify the `ColumnDef` api, but also I don't think the addition of `headerCellProps` is bad, so I'm inclined to leave it.

Reply:
Ok gotcha - sounds good! thanks!